src/link/MachO/Dylib.zig

```diff
@@ -82,22 +123,22 @@ pub fn parse(self: *Dylib) !void {
         return error.MalformedDylib;
     }
 
-    const this_arch: std.Target.Cpu.Arch = switch (self.header.?.cputype) {
-        macho.CPU_TYPE_ARM64 => .aarch64,
-        macho.CPU_TYPE_X86_64 => .x86_64,
-        else => |value| {
-            log.err("unsupported cpu architecture 0x{x}", .{value});
-            return error.UnsupportedCpuArchitecture;
+    const this_arch: std.Target.Cpu.Arch = decodeArch(self.header.?.cputype) catch |err| switch (err) {
+        error.UnsupportedCpuArchitecture => |e| {
+            log.err("unsupported cpu architecture 0x{x}", .{self.header.?.cputype});
```

Comment:
Could you move this into `decodeArch`? Then, you can simplify it to `try decodeArch` without the need to switch on the error variants.

Reply:
That is what it was originally - but I moved it here because a fat lib containing an unsupported arch itself to me doesn't feel like it should cause an error (because we only care about it supporting the target arch at all) . It's only an error on normal dylib.

Reply:
This will always be an error in zld since we most likely wont support anything else in the near future. Also, this helper should really be reused all over the MachO linker but that's an exercise for the future.

Reply:
Sorry, maybe I should be less cautious about this. But isn't it within the bounds of possibility that a certain library might only be available/distributed as a fat lib containing older architectures (like PowerPC) & x86 - we still want ZLD to link against the x86 part of that right without complaining about the fact that it contains another arch within it (this could also hypothetically extend to the future).

Maybe you are right though in that pragmatically - Apple are (well, Apple) and working with fat libs that contain any arch other than ARM64 & x86-64 is not going to happen for the foreseeable future (and fat lib support will become less relevant anyway as Apple Silicon becomes more prevalent) - and older fat libs that were created at previous transition points that can't be rebuilt from source are probably more than old enough to not warrant needing to support.

Reply:
Ah, I think I see what you mean: this would print a log message to the stderr when iterating over the archs contained within the fat lib which you are right, we do not want to do. If someone packs powerpc, x86, in addition to arm64 or x86_64, then that's fine as long as we find a lib we can link against. Nvm then, ignore my comment.